src/main/java/uk/gov/companieshouse/api/strikeoffobjections/chips/ChipsKafkaClient.java

```diff
@@ -1,69 +1,112 @@
+    private String chipsRestInterfacesEndpoint;
+
     @Autowired
-    ApiLogger logger;
+    private ApiLogger logger;
+
+    @Autowired
+    private Supplier<LocalDateTime> dateTimeSupplier;
 
     @Override
-    public void sendToChips(String requestId, ChipsRequest chipsRequest) {
-        Map<String, Object> map = new HashMap<>();
-        map.put("Hello there", "testing");
-        EmailContent emailContent = new EmailContent.Builder().withData(map).withCreatedAt(LocalDateTime.now())
-                .withMessageId("MESSAGE_ID")
-                .withEmailAddress("EMAIL_ADDRESS")
-                .withMessageType("MESSAGE_TYPE")
-                .withOriginatingAppId("strike-off-api")
-                .build();
-        logger.infoContext(requestId, "About to send to chips via kafka");
+    public void sendToChips(String requestId, ChipsRequest chipsRequest) throws ServiceException {
         try {
+            Long timestamp = dateTimeSupplier.get().atZone(ZoneId.systemDefault()).toEpochSecond();
+            ChipsRestInterfacesSend chipsRestInterfacesSend = getChipsRestInterfacesSend(chipsRequest, timestamp);
+
+            Map<String, Object> dataForInfoLogMessage = new HashMap<>();
+            dataForInfoLogMessage.put("topic", chipsRestInterfacesSendTopic);
+            dataForInfoLogMessage.put("message_id", chipsRestInterfacesSend.getMessageId());
+
+            String logMessageSendText = "About to send kafka message to Chips Rest Interfaces Consumer";
+            logger.infoContext(requestId, logMessageSendText, dataForInfoLogMessage);
+
+            Map<String, Object> dataForDebugLogMessage = new HashMap<>(dataForInfoLogMessage);
+            dataForDebugLogMessage.put("message_contents", chipsRestInterfacesSend);
+            logger.debugContext(requestId, logMessageSendText, dataForDebugLogMessage);
+
             Message message = new Message();
-            byte[] serializedData = avroSerializer.serialize(emailContent, schema);
+            byte[] serializedData = avroSerializer.serialize(chipsRestInterfacesSend);
             message.setValue(serializedData);
-            message.setTopic("chips-rest-interfaces-send");
-            message.setTimestamp(emailContent.getCreatedAt().atZone(ZoneId.systemDefault()).toEpochSecond());
+            message.setTopic(chipsRestInterfacesSendTopic);
+            message.setTimestamp(timestamp);
+
             Future<RecordMetadata> future = producer.sendAndReturnFuture(message);
             future.get();
+
+            logger.infoContext(requestId,
+                    "Finished sending kafka message to Chips Rest Interfaces Consumer",
+                    dataForInfoLogMessage);
         } catch (IOException | ExecutionException e) {
             logger.errorContext(requestId, e);
-            return;
-            //throw new ServiceException(e.getMessage());
+            throw new ServiceException(e.getMessage());
```

Comment:
for both of these throw service exceptions, i'd suggest
throw new ServiceException(message, e); so the root cause is still included in the exception

Reply:
yes you're right, I've updated these now and the test